graviton/blackbox.py

```diff
@@ -578,16 +639,30 @@ def execute_one_dryrun(
         teal: str,
         args: Sequence[PyTypes],
         mode: ExecutionMode,
+        abi_method_signature: Optional[str] = None,
         abi_argument_types: Optional[List[Optional[abi.ABIType]]] = None,
         abi_return_type: Optional[abi.ABIType] = None,
         txn_params: dict = {},
         accounts: List[DryRunAccountType] = [],
+        verbose: bool = False,
     ) -> "DryRunInspector":
         assert (
             len(ExecutionMode) == 2
         ), f"assuming only 2 ExecutionMode's but have {len(ExecutionMode)}"
         assert mode in ExecutionMode, f"unknown mode {mode} of type {type(mode)}"
         is_app = mode == ExecutionMode.Application
+
+        if abi_method_signature:
```

Comment:
@tzaffi Leaving a note to mark an out-of-band discussion:  Can `abi_method_signature` replace `abi_argument_types` and `abi_return_type`

* I don't feel _too_ strongly and am willing to accept as is.  My subjective concern is that graviton's APIs will become increasingly difficult to maintain + use due to the growing number of optional parameters.  
* In this particular case, I'm wondering if `abi_method_signature` can replace the discrete args + return type parameters.

Reply:
To clarify current state - @tzaffi mentioned he'll take a look at usage to gauge if it's a _reasonable_ change to consider.

Reply:
I've looked a bit at the usage patterns. My findings are:
* `PyTeal` - moving to usage of an `abi_method_signature` parameter would be straightforward and quick. For the case when we want to dry run an `ars: ABIReturnSubroutine` (the only applicable case), this simply amounts to passing forward `ars.method_signature()` everywhere instead or the argument and/or return ABI types.
* `Graviton` - this is a bit more complicated because we need to pass through the method signature and remove the other parameters in the following methods:
  1. `execute_one_dryrun`
  2. `dryrun_logicsig`
  3. `dryrun_app`
  4. `dryrun_logicsig_on_sequence`
  5. `dryrun_app_on_sequence`
  6. `dryrun_multiapps_on_sequence`

Reply:
## Upshot

Following up with #40